--- a/src/libcamera/egl.cpp
+++ b/src/libcamera/egl.cpp
@@ -180,7 +180,7 @@
 		drm_format = DRM_FORMAT_R8;
 		break;
 	case GL_RG:
-		drm_format = DRM_FORMAT_RG88;
+		drm_format = DRM_FORMAT_R16;
 		break;
 	case GL_RGBA:
 		drm_format = DRM_FORMAT_ARGB8888;
@@ -198,6 +198,4 @@
 		EGL_DMA_BUF_PLANE0_FD_EXT, fd,
 		EGL_DMA_BUF_PLANE0_OFFSET_EXT, 0,
 		EGL_DMA_BUF_PLANE0_PITCH_EXT, (EGLint)eglImage.stride_,
-		EGL_DMA_BUF_PLANE0_MODIFIER_LO_EXT, 0,
-		EGL_DMA_BUF_PLANE0_MODIFIER_HI_EXT, 0,
 		EGL_NONE,
